src/main/java/com/societegenerale/commons/plugin/rules/NoPublicFieldRuleTest.java

```diff
@@ -0,0 +1,44 @@
+package com.societegenerale.commons.plugin.rules;
+
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.fields;
+
+import com.societegenerale.commons.plugin.utils.ArchUtils;
+import com.tngtech.archunit.core.domain.JavaField;
+import com.tngtech.archunit.lang.ArchCondition;
+import com.tngtech.archunit.lang.ConditionEvents;
+import com.tngtech.archunit.lang.SimpleConditionEvent;
+
+import javassist.Modifier;
+
+/**
+ * It is important to respect encapsulation.
+ * 
+ * @see <a href=
+ *      "https://en.wikipedia.org/wiki/Encapsulation_(computer_programming)">Encapsulation</a>
+ */
+
+public class NoPublicFieldRuleTest implements ArchRuleTest {
+	protected static final String NO_PUBLIC_FIELD_VIOLATION_MESSAGE = "Respect encapsulation : no public field";
+
+	@Override
+	public void execute(String path) {
+		fields().should(notBePublic()).check(ArchUtils.importAllClassesInPackage(path, SRC_CLASSES_FOLDER));
```

Comment:
Is there any reason not to use the default ArchUnit API?

```
fields().should().notBePublic()
```

This would also have the advantage of 

* independence of the classpath (which you don't have, if you use `field.reflect()`)
* according to my observations ca. x2 performance gain

Reply:
@codecholeric , I'll try it.